core/src/kura.rs

```diff
@@ -536,9 +414,53 @@ pub struct BlockIndex {
     pub length: u64,
 }
 
-/// Operations available both on locked and unlocked store,
-/// i.e. only read operations.
-impl<L: Lock> BlockStore<L> {
+pub enum LockStatus {
+    Locked,
+    Unlocked,
+}
+
+impl BlockStore {
+    /// Create a new read-only block store in `path`.
+    pub fn new(path: &Path, already_locked: LockStatus) -> Self {
+        if matches!(already_locked, LockStatus::Unlocked) {
+            let path = path.join(LOCK_FILE_NAME);
+            if let Err(e) = fs::File::options()
+                .read(true)
+                .write(true)
+                .create_new(true)
+                .open(path.clone())
+            {
+                match e.kind() {
+                    _ => Err(Error::IO(e, path)),
```

Comment:
I am probable blind, but match arms evaluation depends on order ([example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=10a0a792c96012f37cde12e27528ee01)). I wonder why rustc doesn't print a warning about unreachable arms here.

Reply:
> I wonder why rustc doesn't print a warning about unreachable arms here.

it does i've checked